inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java

```diff
@@ -0,0 +1,172 @@
+import org.apache.inlong.manager.common.util.CommonBeanUtils;
+import org.apache.inlong.manager.common.util.JsonUtils;
+import org.apache.inlong.manager.dao.entity.DataNodeEntity;
+import org.apache.inlong.manager.dao.entity.StreamSinkEntity;
+import org.apache.inlong.manager.dao.mapper.DataNodeEntityMapper;
+import org.apache.inlong.manager.dao.mapper.StreamSinkEntityMapper;
+import org.apache.inlong.manager.pojo.node.cls.ClsDataNodeDTO;
+import org.apache.inlong.manager.pojo.sink.SinkInfo;
+import org.apache.inlong.manager.pojo.sink.cls.ClsSinkDTO;
+import org.apache.inlong.manager.service.resource.sink.SinkResourceOperator;
+import org.apache.inlong.manager.service.sink.StreamSinkService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ClsResourceOperator implements SinkResourceOperator {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ClsResourceOperator.class);
+
+    @Autowired
+    private DataNodeEntityMapper dataNodeEntityMapper;
+    @Autowired
+    private StreamSinkService sinkService;
+    @Autowired
+    private StreamSinkEntityMapper streamSinkEntityMapper;
+
+    @Override
+    public Boolean accept(String sinkType) {
+        return SinkType.CLS.equals(sinkType);
+    }
+
+    @Override
+    public void createSinkResource(SinkInfo sinkInfo) {
+        LOG.info("begin to create sink resources sinkId={}", sinkInfo.getId());
+        if (SinkStatus.CONFIG_SUCCESSFUL.getCode().equals(sinkInfo.getStatus())) {
+            LOG.warn("sink resource [" + sinkInfo.getId() + "] already success, skip to create");
+            return;
+        } else if (InlongConstants.DISABLE_CREATE_RESOURCE.equals(sinkInfo.getEnableCreateResource())) {
+            LOG.warn("create resource was disabled, skip to create for [" + sinkInfo.getId() + "]");
+            return;
+        }
+        this.createTopicID(sinkInfo);
+    }
+
+    private void createTopicID(SinkInfo sinkInfo) {
+        ClsDataNodeDTO clsDataNode = getClsDataNode(sinkInfo);
+        ClsSinkDTO clsSinkDTO = JsonUtils.parseObject(sinkInfo.getExtParams(), ClsSinkDTO.class);
+        try {
+            ClsClient client = getClsClient(clsDataNode);
+            CreateTopicRequest req = getCreateTopicRequest(clsDataNode, clsSinkDTO);
+            CreateTopicResponse resp = client.CreateTopic(req);
+            LOG.info("create cls topic {} success ,topicId {}", clsSinkDTO.getTopicName(), resp.getTopicId());
+            clsSinkDTO.setTopicId(resp.getTopicId());
+            sinkInfo.setExtParams(JsonUtils.toJsonString(clsSinkDTO));
+            StreamSinkEntity streamSinkEntity = new StreamSinkEntity();
+            CommonBeanUtils.copyProperties(sinkInfo, streamSinkEntity, true);
+            streamSinkEntityMapper.updateByIdSelective(streamSinkEntity);
+            this.createTopicIndex(sinkInfo);
```

Comment:
```suggestion
            this.createTopicIndex(sinkInfo);
             CommonBeanUtils.copyProperties(sinkInfo, streamSinkEntity, true);
            streamSinkEntityMapper.updateByIdSelective(streamSinkEntity);
```

Reply:
create an index,  must first fill in the topicId